instrumentation/okhttp/okhttp-2.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/okhttp/v2_2/TracingInterceptor.java

```diff
@@ -5,34 +5,56 @@
 
 package io.opentelemetry.javaagent.instrumentation.okhttp.v2_2;
 
-import static io.opentelemetry.javaagent.instrumentation.okhttp.v2_2.OkHttpClientTracer.tracer;
-
 import com.squareup.okhttp.Interceptor;
 import com.squareup.okhttp.Request;
 import com.squareup.okhttp.Response;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
 import java.io.IOException;
 
 public class TracingInterceptor implements Interceptor {
+  private final Instrumenter<Request, Response> instrumenter;
+  private final ContextPropagators propagators;
+
+  public TracingInterceptor(
+      Instrumenter<Request, Response> instrumenter, ContextPropagators propagators) {
+    this.instrumenter = instrumenter;
+    this.propagators = propagators;
+  }
+
   @Override
   public Response intercept(Chain chain) throws IOException {
     Context parentContext = Context.current();
-    if (!tracer().shouldStartSpan(parentContext)) {
-      return chain.proceed(chain.request());
+    Request request = chain.request();
+
+    if (!instrumenter.shouldStart(parentContext, request)) {
+      return chain.proceed(request);
     }
 
-    Request.Builder requestBuilder = chain.request().newBuilder();
-    Context context = tracer().startSpan(parentContext, chain.request(), requestBuilder);
+    Context context = instrumenter.start(parentContext, request);
+    request = injectContextToRequest(request, context);
 
     Response response;
     try (Scope ignored = context.makeCurrent()) {
-      response = chain.proceed(requestBuilder.build());
+      response = chain.proceed(request);
     } catch (Exception e) {
-      tracer().endExceptionally(context, e);
+      instrumenter.end(context, request, null, e);
       throw e;
     }
-    tracer().end(context, response);
+    instrumenter.end(context, request, response, null);
     return response;
   }
+
+  // Context injection is being handled manually for a reason: we want to use the OkHttp Request
+  // type for additional AttributeExtractors provided by the user of this library
+  // thus we must use Instrumenter<Request, Response>, and Request is immutable
+  private Request injectContextToRequest(Request request, Context context) {
```

Comment:
copied from okhttp3 instrumentation

Reply:
Okhttp2 doesn't have library instrumentation, only javaagent - can you remove the part that mentions library instrumentation from the comment?